core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -67,24 +82,61 @@ oppia.directive('topNavigationBar', [
-            angular.element(evt.currentTarget).parent().addClass('open');
-            $scope.profileDropdownIsActive = true;
-          };
-          $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
-            $scope.profileDropdownIsActive = false;
-          };
-          $scope.onMouseoutDropdownMenuAbout = function(evt) {
-            angular.element(evt.currentTarget)[0].blur();
+          /**
+           * opens the submenu.
+           * @param {object} evt
+           * @param {String} menu - name of menu, on which
+           * open/close action to be performed (aboutMenu,profileMenu).
+           */
+          $scope.openSubmenu = function(evt, menu) {
+            //focus the current target before opening it's submenu.
+            angular.element(evt.currentTarget).focus();
+            $scope.activeMenu = menu;
           };
-          $scope.onMouseoverDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
+          $scope.closeSubmenu = function(evt) {
+            $scope.activeMenu = '';
+            if (evt.currentTarget.nodeName === 'UL') {
+              if (angular.element(evt.currentTarget).closest('li')
+                .find('a').is(':focus')) {
+                angular.element(evt.currentTarget).closest('li')
+                  .find('a').blur();
+              }
+            }
           };
-          $scope.onMouseoutDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
+          /**
+           * Handles keydown events on menus.
+           * @param {object} evt
+           * @param {String} menu - name of menu to perform action 
+           * on(aboutMenu/profileMenu)
+           * @param {object} actions - Map keyboard events('Enter') to
+           * corresponding actions to be performed(open/close).
+           * 
+           * @example
+           *  onMenuKeypress($event, 'aboutMenu', {enter: 'open'})
+           */
+          $scope.onMenuKeypress = function(evt, menu, eventsTobeHandled) {
+            var targetEvents = Object.keys(eventsTobeHandled);
+            for (var i = 0; i < targetEvents.length; i++) {
+              if ($scope.KEYBOARD_EVENT_TO_KEY_CODES[targetEvents[i]]
+                .keyCode === evt.keyCode && evt.shiftKey === 
+                $scope.KEYBOARD_EVENT_TO_KEY_CODES[targetEvents[i]]
+                  .isShiftKeyPressed) {
+                if (eventsTobeHandled[targetEvents[i]] === 'open') {
+                  $scope.openSubmenu(evt, menu);
+                } else {
+                  $scope.closeSubmenu(evt);
+                }
+              }
+            }
           };
+          //Close the submenu if focus or click occurs anywhere outside of 
+          //the menu or outside of it's parent(which opens submenu on hover).
```

Comment:
it's --> its. Add space before "(".

(Note that "it's" means "it is", which isn't appropriate here.)